src/context.ts

```diff
@@ -73,10 +73,22 @@ export class Context {
   public get issue () {
     const payload = this.payload
 
-    return {
-      ...this.repo,
-      number: (payload.issue || payload.pull_request || payload).number
+    const context: { [k: string]: any } = {
+      ...this.repo
     }
+
+    if (payload.issue) {
+      // If it's an issue, the API expects issue_number
+      context.issue_number = payload.issue.number
+    } else if (payload.pull_request) {
+      // If it's a PR, the API expects pull_number
+      context.pull_number = payload.pull_request.number
+    } else {
+      // Otherwise default to the old behaviour
+      context.number = payload.number
+    }
+
```

Comment:
I think we should add this for backwards compat (with a comment explaining that):

```js
context.number = context.number || context.issue_number || context.pull_number
```

Otherwise this'll be a breaking change and require a major version bump right now, and we can ease into a bit - and get rid of this in the next major version.